src/collections/vec.rs

```diff
@@ -1778,12 +1778,43 @@ impl<'bump, T: 'bump + Clone> Vec<'bump, T> {
 }
 
 impl<'bump, T: 'bump + Copy> Vec<'bump, T> {
+    /// Helper method to copy all of the items in `other` and append them to the end of `self`.
+    ///
+    /// SAFETY:
+    ///   * The caller is responsible for:
+    ///       * calling `reserve` beforehand to guarantee that there is enough capacity to
+    ///         store `other.len()` more items.
+    ///       * updating the length of the `Vec` afterward to reflect the number of items added.
+    ///   * The caller must also guarantee that `self` and `other` do not overlap.
+    unsafe fn extend_from_slice_copy_unchecked(&mut self, other: &[T]) {
+        let old_len = self.len();
+        debug_assert!(old_len + other.len() <= self.capacity());
+
+        // SAFETY:
+        // * `src` is valid for reads of `other.len()` values by virtue of being a `&[T]`.
+        // * `dst` is valid for writes of `other.len()` bytes because the caller of this
+        //   method is required to `reserve` capacity to store at least `other.len()` items
+        //   beforehand.
+        // * Because `src` is a `&[T]` and dst is a `&[T]` within the `Vec<T>`,
+        //   `copy_nonoverlapping`'s alignment requirements are met.
+        // * Caller is required to guarantee that the source and destination ranges cannot overlap
+        unsafe {
+            let src = other.as_ptr();
+            let dst = self.as_mut_ptr().add(old_len);
+            ptr::copy_nonoverlapping(src, dst, other.len());
+        }
+    }
```

Comment:
This function doesn't update `self.len`, but I think it should, no?

Reply:
This is the caller's responsibility, which allows this method to be used for both the single-slice and multi-slice cases. There's a note about this in the doc comment above:

```rust
/// * The caller is responsible for:
///   * calling `reserve` beforehand to guarantee that there is enough capacity to
///     store `other.len()` more items.
///   * updating the length of the `Vec` afterward to reflect the number of items added.
```

I'll add an explicit note about calling `set_len` to that last bullet. Please let me know if you'd like me to do something else to make it clearer.